### ISSUE TITLE
sparse tensor creation should raise IndexError if indices are out of bounds

### ISSUE THREAD

**Issue by vmichals** [pytorch/pytorch#5748]
I think the following statement should raise an IndexError, indicating index 2 for axis 0 is out of bounds:
```
torch.sparse.FloatTensor(torch.LongTensor([[1, 2],[0, 2]]), torch.FloatTensor([.5,.5]), torch.Size([2,3])).to_dense()
```
At the moment it just creates the tensor in the requested size, ignoring the elements that are beyond the limits. This could introduce some bugs, that are hard to track down.

**Comment by vmichals** [comment]
already fixed by #4113. Seems I have to build from source.

**Comment by zou3519** [comment]
I'm not 100% sure this is fixed. I'm running the latest master and running your code above isn't resulting in an error. Something probably regressed somewhere.

**Comment by zou3519** [comment]
In addition, this check doesn't exist for the CUDA sparse tensor constructor

**Comment by vmichals** [comment]
Thanks for checking! I just assumed it was fixed, because #4113 seemed to address at least something related. Next time I'll double check before closing my issue!
